### PR TITLE
Added method to remove email from profile

### PR DESCRIPTION
**Pull request by vytis**
## What's in this PR?

Changes needed to support email removal from UI. Added a method that sends a request to delete the email and some basic tests for it.

### CODE REVIEW

File: Source/Synchronization/Strategies/UserProfileUpdateRequestStrategy.swift [hunk 1]
```diff
@@ -205,6 +216,16 @@ extension UserProfileRequestStrategy : ZMSingleRequestTranscoder {
                 self.userProfileUpdateStatus.didFailEmailUpdate(error: error)
             }
             
+        case self.emailDeleteSync:
+            if response.result == .success {
+                ZMUser.selfUser(in: managedObjectContext).needsToBeUpdatedFromBackend = true
```
Comment (marcoconti83) [inline Source/Synchronization/Strategies/UserProfileUpdateRequestStrategy.swift]: Why do we need to redownload the user here? Are you afraid that just deleting the email from the Core Data user would be an issue if someone had already set a new email on another device?

Comment (vytis) [inline Source/Synchronization/Strategies/UserProfileUpdateRequestStrategy.swift]: I guess it's not really needed in this case. The initial problem was that email deletion event doesn't come from the backend in the same way as editing email does. So I ended up removing email in the UI project from `ZMUser` after request is finished.

Comment (marcoconti83) [inline Source/Synchronization/Strategies/UserProfileUpdateRequestStrategy.swift]: I would make that change at the sync-engine level, to abstract away from the UI the fact that you need to do certain operations on the local data once you get a certain callback.

**Comment by vytis** [comment]
@marcoconti83 removing email in the sync engine

**Comment by vytis** [comment]
It has some UX issues so we will not be implementing email removal for now, closing this PR.

**Pull request closed without merging**